Cargo.lock

```diff
@@ -1455,10 +1490,24 @@ version = "0.21.0"
 source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "d557201d71792487bd2bab637ab5be9aa6fff59b88e25e12de180b0f9d2df60f"
 dependencies = [
- "ra-ap-rustc_index",
+ "ra-ap-rustc_index 0.21.0",
  "ra-ap-rustc_lexer",
 ]
 
+[[package]]
+name = "ra-ap-rustc_pattern_analysis"
+version = "0.33.0"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "6c4085e0c771fd4b883930b599ef42966b855762bbe4052c17673b3253421a6d"
+dependencies = [
+ "derivative",
```

Comment:
Oof, we went through some pains to get rid of `syn` 1.0. But you do seem to use it in a couple of places, so doing that by hand would probably be too annoying.

Reply:
Oh no 🙈 I mean it's not *too* hard to impl by hand

Reply:
It's not, but I don't want to force you to spend time on that right now. It's okay to merge, then clean it up later (ideally).

Reply:
We should just fix up the assist to inline custom proc-macro derive expansions (which currently only supports builtin derives) :)

But to chime in as well, I'd like to see derivative not stick in our dependency tree for long. Iirc it, like syn, has a pretty bad compile time impact. 

> It's okay to merge, then clean it up later (ideally).

☝️

Reply:
Maybe https://github.com/mcarton/rust-derivative/pull/118 can save us! If it ever gets merged

Reply:
It seems rustc also wants to get rid of syn 1.0 https://github.com/rust-lang/rust/issues/109302

Reply:
Hmm, actually rust-derivative does not seem to be maintained anymore (3 years since the last commit). Anyways, since rustc wants to bump to syn 2 as well T-compiler will probably figure something out there at some point.

Reply:
Yeah, that PR I linked was opened recently by the person who's working on moving rustc to syn2, hopefully they'll take some kind of action

Reply:
Fine, I'll do it :joy:.